src/dev/LCD.cpp

```diff
@@ -23,71 +39,221 @@ void LCD::commandWrite(uint8_t data) {
-        this->commandWrite(0x00);      //column address lower 4 bits + 0x00
-        for (int j = 0; j < 128; j++) {//128 columns wide
-            this->dataWrite(0x00);     //write clear pixels
-            bitMap++;
+void LCD::clearLCD() {
+    uint8_t i, j;
+    unsigned char page = 0xB0;// The starting page
+
+    this->commandWrite(0xAE);              // Display OFF
+    this->commandWrite(0x40);              // Display start address + 0x40
+    for (i = 0; i < 8; i++) {              // 64 pixel display / 8 pixels per page = 8 pages
+        this->commandWrite(page);          // Send page address
+        this->commandWrite(0x10);          // Column address upper 4 bits + 0x10
+        this->commandWrite(0x00);          // Column address lower 4 bits + 0x00
+        for (j = 0; j < screenSizeX; j++) {// 128 columns wide
+            this->dataWrite(0x00);         // Write clear pixels
         }
-        page++;//after 128 columns, go to next page
+        page++;// After 128 columns, go to next page
     }
     this->commandWrite(0xAF);
 }
 
+void LCD::clearArea(uint8_t width, uint8_t height, uint8_t page, uint8_t column) {
+    uint8_t i, j;
+    // Calculate the correct column address from an 8-bit number.
+    // The issue is that the display takes 4 bit numbers, instead of 8.
+    // So you need to split it into two 4 bit numbers.
+    uint8_t columnUpperAddress = column;
+    columnUpperAddress >>= 4;
+
+    uint8_t columnLowerAddress = column;
+    columnLowerAddress <<= 4;
+    columnLowerAddress >>= 4;
+
+    uint8_t amountOfPages = height / 8;
+    if (height < 8) {
+        amountOfPages = 1;
+    }
+
+    this->commandWrite(0xAE);                         // Display OFF
+    this->commandWrite(0x40);                         // Display start address + 0x40
+    for (i = 0; i < amountOfPages; i++) {             // 64 pixel display / 8 pixels per page = 8 pages
+        this->commandWrite(0xB0 + page);              // Send page address
+        this->commandWrite(0x10 + columnUpperAddress);// Column address upper 4 bits + 0x10
+        this->commandWrite(0x00 + columnLowerAddress);// Column address lower 4 bits + 0x00
+
+        for (j = 0; j < width; j++) {
+            this->dataWrite(0x00);// Write Clear Pixels
+        }
+
+        page++;// After 128 columns, go to next page
+    }
+    this->commandWrite(0xAF);// Finish Writing
+}
+
 void LCD::displayMap(const uint8_t* bitMap) {
+    uint8_t i, j;
     uint8_t page = 0xB0;
-    this->commandWrite(0xAE);          //Display OFF
-    this->commandWrite(0x40);          //Display start address + 0x40
-    for (int i = 0; i < 8; i++) {      //64 pixel display / 8 pixels per page = 8 pages
-        this->commandWrite(page);      //send page address
-        this->commandWrite(0x10);      //column address upper 4 bits + 0x10
-        this->commandWrite(0x00);      //column address lower 4 bits + 0x00
-        for (int j = 0; j < 128; j++) {//128 columns wide
```

Comment:
Same thing mentioned previously. Sort of curious why you do it this way?

Reply:
If I remember correctly this was just the way it was when I got to the code and I never changed it. I moved them all into the loops.